csrc/quantization/cutlass_w8a8/moe/moe_data.cu

```diff
@@ -101,3 +101,35 @@ void get_cutlass_moe_mm_data_caller(
       static_cast<int32_t*>(atomic_buffer.data_ptr()), topk_ids.numel(),
       topk_ids.size(1));
 }
+
+__global__ void compute_pplx_data(
+    int32_t* expert_offsets, int32_t* problem_sizes1, int32_t* problem_sizes2,
+    const int32_t* __restrict__ expert_num_tokens,
+    const int64_t* __restrict__ non_zero_expert_idxs, const int padded_m,
+    const int n, const int k) {
+  int expert_idx_out = threadIdx.x;
+  int expert_idx_in = non_zero_expert_idxs[expert_idx_out];
+  expert_offsets[expert_idx_out] = expert_idx_in * padded_m;
```

Comment:
```suggestion
  int expert_idx_in = static_cast<int32_t>(non_zero_expert_idxs[expert_idx_out]);
  expert_offsets[expert_idx_out] = expert_idx_in * padded_m;
```

Could you doublecheck and try to not add any warnings to the build? (The implicit down-conversion here looks safe enough to me, but best to avoid implicit conversions)

Reply:
I don't see this producing any new warnings, I'll make the change